docs/source/library-user-guide/extending-sql.md

```diff
@@ -0,0 +1,339 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Extending DataFusion's SQL Syntax
+
+DataFusion provides a flexible extension system that allows you to customize SQL
+parsing and planning without modifying the core codebase. This is useful when you
+need to:
+
+- Support custom operators from other SQL dialects (e.g., PostgreSQL's `->` for JSON)
+- Add custom data types not natively supported
+- Implement SQL constructs like `TABLESAMPLE`, `PIVOT`/`UNPIVOT`, or `MATCH_RECOGNIZE`
+
+## Architecture Overview
+
+When DataFusion processes a SQL query, it goes through these stages:
+
+```text
```

Comment:
This is a great flow chart ❤️ 

One thing I noticed it is is slightly clipped :
<img width="819" height="576" alt="Image" src="https://github.com/user-attachments/assets/31fcf2d0-f52f-49f6-93f4-d2a7b79c8c1c" />

If we were able to tighten up the SqlToRel box (maybe make it two lines) it might fit